src/test/bridges/notional/NotionalUnit.t.sol

```diff
@@ -0,0 +1,318 @@
+    address public constant ETH = address(0);
+    address public constant CETH = address(0x4Ddc2D193948926D02f9B1fE9e1daa0718270ED5);
+    address public constant DAI = address(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+    address public constant CDAI = address(0x5d3a536E4D6DbD6114cc1Ead35777bAB948E3643);
+    address public constant WETH = address(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+    address public constant USDC = address(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48);
+    address public constant CUSDC = address(0x39AA39c021dfbaE8faC545936693aC917d5E7563);
+    address public constant WBTC = address(0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599);
+    address public constant CWBTC = address(0xccF4429DB6322D5C611ee964527D42E5d685DD6a);
+    mapping(address => uint16) public currencyId;
+    address[8] public tokens;
+    uint256[8] public lowerBound;
+    uint256[8] public upperBound;
+    mapping(address => address) public underlyingTokenToAssetToken;
+    mapping(address => address) public assetTokenToUnderlyingToken;
+    mapping(address => bool) public underlyingToken;
+    IWrappedfCashFactory public constant FCASH_FACTORY =
+        IWrappedfCashFactory(0x5D051DeB5db151C2172dCdCCD42e6A2953E27261);
+    AztecTypes.AztecAsset private empty;
+    NotionalBridgeContract public bridge;
+    NotionalViews public notionalView = NotionalViews(0x1344A36A1B56144C3Bc62E7757377D288fDE0369);
+
+    function setUp() public {
+        bridge = new NotionalBridgeContract(address(ROLLUP_PROCESSOR));
+        // Approve ERC20 tokens for ROLLUP_PROCESSOR
+        tokens[0] = ETH;
+        tokens[1] = CETH;
+        tokens[2] = DAI;
+        tokens[3] = CDAI;
+        tokens[4] = USDC;
+        tokens[5] = CUSDC;
+        tokens[6] = WBTC;
+        tokens[7] = CWBTC;
+        currencyId[ETH] = 1;
+        currencyId[CETH] = 1;
+        currencyId[DAI] = 2;
+        currencyId[CDAI] = 2;
+        currencyId[USDC] = 3;
+        currencyId[CUSDC] = 3;
+        currencyId[WBTC] = 4;
+        currencyId[CWBTC] = 4;
+        lowerBound[0] = 1e16;
+        lowerBound[1] = 1e8;
+        lowerBound[2] = 1e16;
+        lowerBound[3] = 1e9;
+        lowerBound[4] = 1e6;
+        lowerBound[5] = 1e9;
+        lowerBound[6] = 1e5;
+        lowerBound[7] = 1e6;
+        upperBound[0] = 1e20;
+        upperBound[1] = 1e12;
+        upperBound[2] = 1e22;
+        upperBound[3] = 1e15;
+        upperBound[4] = 1e12;
+        upperBound[5] = 1e15;
+        upperBound[6] = 1e9;
+        upperBound[7] = 1e10;
+        underlyingTokenToAssetToken[ETH] = CETH;
+        underlyingTokenToAssetToken[WETH] = CETH;
+        underlyingTokenToAssetToken[DAI] = CDAI;
+        underlyingTokenToAssetToken[WBTC] = CWBTC;
+        underlyingTokenToAssetToken[USDC] = CUSDC;
+        assetTokenToUnderlyingToken[CETH] = ETH;
+        assetTokenToUnderlyingToken[CUSDC] = USDC;
+        assetTokenToUnderlyingToken[CWBTC] = WBTC;
+        assetTokenToUnderlyingToken[CDAI] = DAI;
```

Comment:
You don't need this mapping because all the cTokens have underlying method on them which gives you the address.